### PR TITLE
Set fixed height on NavigationView icon ViewBox

### LINKED ISSUE #822: Rendering bugs with NavigationView across different releases.
The NavigationView control exhibits inconsistent rendering across different releases of Microsoft.UI.Xaml.Controls.

Example with Microsoft.UI.Xaml v 2.0.181018004, you get this which is mostly correct:

image

Notice that the XAML markup shown below does not properly extend the selection area:

`<ui:NavigationViewItem.Icon>
<PathIcon Data= ...  />
</ui:NavigationViewItem.Icon>`

When using Microsoft.UI.Xaml v2.1.190405004.2 you get this:

image

Which is way worse and quite wrong.

Please start using a regression testing system that takes UI snapshots to compare results.  This sort of thing should never happen with newer releases.

### PR DESCRIPTION
**Pull request by kaiguo**
Fixes #822

We removed the fixed height on NavigationViewItem to allow dynamic height adjusting, but this makes the NavViewItem icon size inconsistent when using custom ones. Icons will just render at their default sizes since there's no constrains.

Fixing it by giving a fixed height to icon ViewBox.

### CODE REVIEW

**Comment by jevansaks** [comment]
Did you find out the change where this was originally made to see what the scenario was? Does this potentially break one of the density scenarios?

Also we should start getting some XML-based masters established here for catching regressions like this because the NavView scenarios are very complex.

**Comment by kaiguo** [comment]
> Did you find out the change where this was originally made to see what the scenario was? Does this potentially break one of the density scenarios?
> 
> Also we should start getting some XML-based masters established here for catching regressions like this because the NavView scenarios are very complex.

I think it's this https://github.com/microsoft/microsoft-ui-xaml/pull/361. From what I can tell the icon size has always been 16*16. That change removed the constrain by changing Height to MinHeight.

File: dev/NavigationView/NavigationView_rs1_themeresources.xaml [hunk 1]
```diff
@@ -591,7 +591,9 @@
                                 <ColumnDefinition Width="*" />
                             </Grid.ColumnDefinitions>
 
-                            <Viewbox x:Name="IconBox"                                
+                            <Viewbox x:Name="IconBox"
+                                Width="16"
+                                Height="16"
                                 Margin="{ThemeResource NavigationViewItemOnLeftIconBoxMargin}">
```
Comment (chrisglein) [inline dev/NavigationView/NavigationView_rs1_themeresources.xaml]: If you unpack these two numbers: IconColumn = 40, and this Margin = 12,0,12,0, you end up with 40-24=16. So the Width should already be accounted for. Do you need both?

Comment (kaiguo) [reply dev/NavigationView/NavigationView_rs1_themeresources.xaml]: Right, probably don't need width, will remove.

**Comment by mdtauk** [comment]
Instead of hard-coding the values, shouldn't they be Static/Theme Resource values which would allow for easy overriding by developers who may wish to customise the control?

**Comment by jevansaks** [comment]
> Instead of hard-coding the values, shouldn't they be Static/Theme Resource values which would allow for easy overriding by developers who may wish to customise the control?

I think it's reasonable to lift out values that _can_ be customized reasonably well. In this case just overriding these values wouldn't cause the icons to grow so it gives a false sense that it's customizable. If we wanted the icon size to be customizable we'd need to revamp the layout a bit. We shouldn't have the perfect be the enemy of the good, else we spend a lot of time building features that no one uses or don't align with our vision of the control.

**Comment by chrisglein** [comment]
> Instead of hard-coding the values, shouldn't they be Static/Theme Resource values which would allow for easy overriding by developers who may wish to customise the control?
> 
> I think it's reasonable to lift out values that can be customized reasonably well. In this case just overriding these values wouldn't cause the icons to grow so it gives a false sense that it's customizable. If we wanted the icon size to be customizable we'd need to revamp the layout a bit. We shouldn't have the perfect be the enemy of the good, else we spend a lot of time building features that no one uses or don't align with our vision of the control.

I had the same thought reviewing this, then I remembered that that 16 value is directly or indirectly hardcoded throughout the control. If we could promise a level of support for it being customized that'd be good, but if it's not I think it's better to keep the values inline. And since that wasn't the purpose behind the change, I think it's correct to keep the values hardcoded in there.

**Comment by mdtauk** [comment]
Did you make this fixed size a style resource, do it can be overridden?

**Comment by kaiguo** [comment]
> Did you make this fixed size a style resource, do it can be overridden?

I didn't, as Jevan and Chris said above, the width is implicitly hard coded by column width and margin, adding a resource key doesn't really make it customizable. We'd have to revamp the layout if we want to make the size customizable. This PR was just to fix the issue in #822 and make the icon size back to what it originally was, so I left it hard coded.

**Pull request merged**